llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -2909,13 +2909,6 @@ void VPlanTransforms::canonicalizeEVLLoops(VPlan &Plan) {
 void VPlanTransforms::replaceSymbolicStrides(
     VPlan &Plan, PredicatedScalarEvolution &PSE,
     const DenseMap<Value *, const SCEV *> &StridesMap) {
-  // Replace VPValues for known constant strides guaranteed by predicate scalar
```

Comment:
This still holds I think, it make ssure we only replace users that are guaranteed to be dominated by the runtime checks we insert.

Reply:
The comment is the explanation for the check below, which I think should be kept to ensure we only replace strides we know will be guarded by runtime checks